libopae/plugins/xfpga/feature_plugins/feature_api.c

```diff
@@ -188,20 +192,22 @@ xfpga_fpgaFeatureEnumerate(fpga_handle handle, fpga_feature_properties *prop,
 
 		if (_ftoken->feature_type == prop->type) {
 			if ((!uuid_is_null(prop->guid) && (uuid_compare(prop->guid, guid) == 0))
-				|| (uuid_is_null(prop->guid))) { 
-				if (*num_matches < max_tokens) {
-					fpga_feature_token tmp = 0;
-					feature_adapter_table * adapter;
+				|| (uuid_is_null(prop->guid))) {
+				if (tokens) {
+					if (*num_matches < max_tokens) {
+						fpga_feature_token tmp = 0;
+						feature_adapter_table * adapter;
 					
-					result = xfpga_fpgaCloneFeatureToken((fpga_feature_token)_ftoken, &tmp);
-					if (result	!= FPGA_OK) {
-						FPGA_MSG("Error cloning token");
-						return result;
+						result = xfpga_fpgaCloneFeatureToken((fpga_feature_token)_ftoken, &tmp);
+						if (result	!= FPGA_OK) {
+							FPGA_MSG("Error cloning token");
+							return result;
+						}
+						adapter = get_feature_plugin_adapter(guid);
+						wrapped_feature_token *wt = 
+							allocate_wrapped_feature_token(tmp, adapter);
+						tokens[*num_matches] = wt;
 					}
-					adapter = get_feature_plugin_adapter(guid);
-					wrapped_feature_token *wt = 
-					     allocate_wrapped_feature_token(tmp, adapter);
-					tokens[*num_matches] = wt;
 				}
 				++(*num_matches);
 			}
```

Comment:
dfh is still not being updated inside the loop body.  This will result in an infinite loop.  Please address.

Reply:
yes. should copy the data at new offiset to the dfh at each loop.

Reply:
How does the code break out of this loop?  The while condition concerns dfh, but dfh isn't being mutated inside the loop.

Reply:
The final DFH's eol should be TRUE which breaks the while loop.

Reply:
Just found out that DFH was updated at line 167 inside the loop.

Reply:
Thanks for clarifying.